### PR TITLE
Support quad-complex ABS intrinsic

### PR DESCRIPTION
**Pull request by wanbinchen-hnc**
1. Add relevant runtime routines.
2. Add relevant ILM opcodes and Fortran runtime symbols.
3. Add ILM generation and lowering logic.
5. Support quad-complex test
4. Add test cases.

### CODE REVIEW

**Comment by wanbinchen-hnc** [comment]
Gentle Ping...

File: runtime/libpgmath/lib/common/mthdecls.h [hunk 1]
```diff
@@ -193,6 +193,8 @@ quad_complex_t pgmath_cmplxl(float128_t r, float128_t i)
         float _f(float real, float imag)
 #define	DBLFUNC_C_(_f)    \
         double _f(double real, double imag)
+#define	QUADFUNC_C_(_f)    \
+        long double _f(long double real, long double imag)
```
Comment (bryanpkc) [inline runtime/libpgmath/lib/common/mthdecls.h:197]: Use `float128_t` instead of `long double`, which is not portable.

File: tools/flang1/flang1exe/symacc.h [hunk 2]
```diff
@@ -16,7 +16,7 @@ struct SYM;
 #include "sharedefs.h"
 
 /* Keep in sync with INIT_NAMES_SIZE, in syminidf.h. */
-#define STB_NSIZE 5069
+#define STB_NSIZE 5075
```
Comment (bryanpkc) [inline tools/flang1/flang1exe/symacc.h:19]: Please also check if a similar change is needed in `tools/flang2/flang2exe/symacc.h`.

File: test/f90_correct/src/check_mod.F90 [hunk 3]
```diff
@@ -933,6 +937,95 @@ subroutine checkc8(reslt, expct, np, atoler, rtoler, ulptoler, ieee)
       endif
       return
     end subroutine checkc8
+    ! complex*32
```
Comment (pawosm-arm) [inline test/f90_correct/src/check_mod.F90:940]: please move this comment inside of the `#ifdef` block

File: tools/flang1/flang1exe/symacc.h (hunk 2 above)
Comment (wanbinchen-hnc) [reply tools/flang1/flang1exe/symacc.h:19]: The `STB_NSIZE` keeps in sync with `INIT_NAMES_SIZE` (both are 3289) in `build/flang/tools/flang2/utils/symtab/syminidf.h`, so the similar change is unneeded.

Comment (bryanpkc) [reply tools/flang1/flang1exe/symacc.h:19]: Thanks for confirming.

File: test/f90_correct/src/check_mod.F90 (hunk 3 above)
Comment (wanbinchen-hnc) [reply test/f90_correct/src/check_mod.F90]: OK, already updated.

File: runtime/libpgmath/lib/common/mthdecls.h (hunk 1 above)
Comment (wanbinchen-hnc) [reply runtime/libpgmath/lib/common/mthdecls.h]: OK, already updated.

**Comment by wanbinchen-hnc** [comment]
@bryanpkc This PR has been updated.

File: test/f90_correct/src/qc_abs_01.f08 [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+!** SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+!*  Test intrinsics function abs with quad precision complex.
+
+program main
+  use check_mod
+  implicit none
+  integer, parameter :: k = 16
+  integer, parameter :: n = 15
+  complex(kind=k), parameter :: t1 = (1.23434590874573456543454E+30_16, &
```
Comment (bryanpkc) [inline test/f90_correct/src/qc_abs_01.f08:12]: Are the test input data randomly selected? Did you include any boundary values such as maximum and minimum quad-precision values? Please add a comment to elaborate on the data selection.

Comment (bryanpkc) [reply test/f90_correct/src/qc_abs_01.f08:12]: This applies to the test data in the other test cases as well.

Comment (wanbinchen-hnc) [reply test/f90_correct/src/qc_abs_01.f08:12]: I have refactored these test cases and PR has been updated, please have a look

Comment (bryanpkc) [reply test/f90_correct/src/qc_abs_01.f08:12]: @wanbinchen-hnc Did you push your updates correctly? When I try to review the differences, GitHub reports that "ae192dc9b254a9eda5d97266a17c3446ee7f667d and fde217fc7b80f0d68f6dbc1ca02c10dba70993f6 are identical".

Comment (wanbinchen-hnc) [reply test/f90_correct/src/qc_abs_01.f08]: Thanks for your reminding, I have pushed my updates again.

**Review by bryanpkc — APPROVED** [review]
LGTM

**Review by pawosm-arm — APPROVED** [review]
